staking.md

```diff
@@ -0,0 +1,606 @@
+       - Must have sufficient AB to cover the increase.
+     - **Minimum Stake Requirement**:
+       - After adjustment, if the participant's stake is below `MINIMUM_STAKE`, they will not be eligible for leader selection in the upcoming cycle.
+   - **Behavior**:
+     - Updates the participant's stake commitment for the next staking cycle.
+     - Recalculats LB:
+       - **LB = amount**
+     - Recalculates AB:
+       - **AB = TB - LB**
+
+
+4. **`getStakedAmount(producer:pubkey, epoch:integer)`**
+   - **Description**: Returns the amount of tokens the participant has committed to stake in the staking cycle starting at `epoch` and ending in `epoch + 3`
+   - **Parameters**:
+     - `producer`: The pubkey of the producer.
+     - `epoch`: The epoch number. (Or possibly a block height?)
+   - **When Callable**: At any time.
+
+5. **`getAvailableBalance(producer:pubkey)`**
+   - **Description**: Returns the participant's current Available Balance (AB).
+   - **Parameters**:
+     - `producer`: The pubkey of the producer.
+   - **When Callable**: At any time.
+
+---
+
+## **Explanation of Balances Across Epochs**
+
+### **Key Concepts**
+
+- **Total Balance (TB)**: Sum of all tokens deposited via `deposit()` and not yet withdrawn.
+- **Locked Balance (LB)**: The total amount of tokens locked in the current staking cycle, i.e., epochs `N` to `N + 3`.
+- **Available Balance (AB)**: Calculated as the difference between TB and LB.
+  - **AB = TB - LB**
+
+### **Lifecycle of Tokens in the Staking Contract**
+
+1. **Staking Tokens (`deposit()`):**
+   - Increases TB and AB.
+   - No immediate effect on staking commitments until `adjustStake()` (or `stake()`) is called.
+
+2. **Adjusting Stake (`adjustStake()`):**
+   - Adjusts the participant's stake commitment for the next staking cycle.
+   - **Positive Adjustment:**
+     - Decreases AB by `amount`.
+     - Increases the stake commitment for the next staking cycle by `amount`.
+   - **Negative Adjustment:**
+     - Increases AB by `-amount`.
+     - Decreases the stake commitment for the next staking cycle by `-amount`.
+   - **AB Recalculation:**
+     - **AB = TB - LB**
+
+3. **Withdrawal (`withdraw()`):**
+   - Decreases TB and AB by `amount`.
+   - Cannot affect tokens locked in the any of the 4 concurrent staking cycles.
+
+4. **Epoch Transition:**
+   - At the end of the Payout Epoch (`N + 3`):
+     - Tokens locked in the current staking cycle become available.
+     - **LB** decreases by the amount of tokens that were locked.
+     - **AB** increases accordingly.
+
+---
+
+### **Updated Example Scenario**
+
+Let's illustrate how the balances change over epochs with this updated model.
+
+#### **Constants and Parameters**
+
+- **Current Epoch (`N`)**: 10 (Staking Epoch)
+- **Staking Cycle Length (`C`)**: 4 epochs (epochs 10 to 13)
+  - Epoch 10: Staking Epoch
+  - Epoch 11: Leader Election Epoch
+  - Epoch 12: Block Production Epoch
+  - Epoch 13: Payout Epoch
```

Comment:
I think it would simplify the discussion if this aligns with the other document and the implementation...